### PR TITLE
Modify MilvusIndex to support Zilliz

### PR DESCRIPTION
**Pull request by filip-halt**
Adding Zilliz support in milvus index and documentation on how to use.

### CODE REVIEW

File: docs/how_to/integrations/vector_stores.md [hunk 1]
```diff
@@ -206,6 +206,7 @@ response = index.query("What did the author do growing up?")
 
 **Milvus Index Construction/Querying**
```
Comment (jerryjliu) [inline docs/how_to/integrations/vector_stores.md:207]: @filip-halt would you like to clarify here too?

**Review by jerryjliu — APPROVED** [review]
this generally looks good! some minor comments though @filip-halt

File: gpt_index/vector_stores/milvus.py [hunk 2]
```diff
@@ -242,9 +246,21 @@ def _create_index(self) -> None:
...
-            self.collection.create_index(
-                "embedding", index_params=self.index_params, using=self.alias
-            )
+            try:
+                self.collection.create_index(
+                    "embedding", index_params=self.index_params, using=self.alias
+                )
+            # If default did not work, most likely on Zilliz Cloud
+            except MilvusException:
+                # Attempt creating autoindex
```
[the remark this answers is not in the document]
Comment (filip-halt) [reply gpt_index/vector_stores/milvus.py:255]: @jerryjliu Unfortunately as of now there is no way to get around the try except as the client doesn't know if its connected to milvus or zilliz and zilliz only accepts AUTOINDEX.

File: docs/how_to/integrations/vector_stores.md (hunk 1 above)
Comment (filip-halt) [reply docs/how_to/integrations/vector_stores.md:207]: Let me do that. Wasnt sure how much I was allowed to add in that section in terms of descriptions and extra info.

**Review by jerryjliu — APPROVED** [review]

**Pull request merged**